packages/contracts/contracts/interfaces/IPriceOracle.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.19;
+
+interface IPriceOracle {
+  /// @notice Initialize oracle for address
+  function initialize(address caller, bytes calldata options) external;
+
+  /// @notice Returns price as FP96 value
+  function getBalancePrice(address caller) external view returns (uint256);
```

Comment:
AFAIS `caller` is always set as `address(this)`, but that value can be obtained by oracle via `msg.sender`. This param should be either renamed or removed.